data/collector.go

```diff
@@ -271,6 +282,48 @@ func NewCollector(captureFunc CaptureFunc, params CollectorParams) (Collector, e
 	}, nil
 }
 
+// PBStructToBSON converts a structpb.Struct to a bson.M.
```

Comment:
nit, i wonder if this fits in collector.go or if there's another "utils"-like package that can host this. ok if that's not the right convention here

Reply:
also, should we have any unit tests coverage for this?

Reply:
done, FYI both the implementation & the tests are copy & paste from https://github.com/viamrobotics/app/blob/main/datasync/bson_utils_test.go & https://github.com/viamrobotics/app/blob/main/datasync/bson_utils.go